FreeRTOS/iot-reference-arm-corstone3xx
# Restructure repository

Description
-----------


In the fast-paced world of software development, maintaining a well-organised and structured code repository is crucial.

This document outlines changes made aimed at enhancing the clarity, consistency, and overall efficiency of our code base.
By adopting standardised naming conventions, reorganising files, and redefining subdirectory structures, we aim to streamline collaboration, simplify maintenance, and make the code base more approachable for both current and future developers. Additionally, the incorporation of clear CMake targets will improve the build process, promoting a more robust and scalable development environment.

These changes are essential for fostering a sustainable and agile development workflow, ultimately leading to increased productivity and a higher quality code base.

Test Steps
-----------


Checklist:
----------



Related Issue
-----------

https://github.com/FreeRTOS/iot-reference-arm-corstone3xx/issues/25

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
